Simple-Projects/Simple_Projects/Basic/List/List.cpp

```diff
@@ -72,28 +64,26 @@ void List::push_back(int value) {
        array = newArray;
     }
     array[current] = value;
-    current = newCurrent;
+	++current;
 }
 
 int List::pop_back() {
     if ( current == 0 ) {
        throw ZeroLenException();
     }
-    current -= 1;
-    return array[current];
+    return array[--current];
 }
 
 void List::sort() {
-    int last = current - 1;
-
-	for ( int i = 0; i < last; i++ ) {
-		for ( int j = 0; j < last; j++ ) {
-			int min = j + 1;
-            if ( array[j] > array[min] ) {
-                int temp = array[j];
-
-                array[j] = array[min];
-                array[min] = temp;
+    const int last = current - 1;
+
+	// Bubble sorting will keep an invariant as follows: [sorted|unsorted]
```

Comment:
Note how thinking on the sort snippet in terms of an invariant makes it easier to understand.
The amount of elementary operations reduced twofold, too.